### PR TITLE
Add support for  influxdb database connection

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
resource/database_secret_backend_connection: Add support for InfluxDB connections
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccDatabaseSecretBackendConnection_influxdb'
--- PASS: TestAccDatabaseSecretBackendConnection_influxdb (11.77s)

```

### CODE REVIEW

**Comment by DrFaust92** [comment]
Hi, any news on this? i see this has be assigned for a while @vinay-gopalan

File: vault/resource_database_secret_backend_connection.go [hunk 1]
```diff
@@ -171,6 +174,77 @@ func databaseSecretBackendConnectionResource() *schema.Resource {
...
+							Sensitive:    true,
+							ValidateFunc: validation.StringIsJSON,
+						},
+						"connect_timeout": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     5,
+							Description: "The number of seconds to use as a connection timeout.",
+						},
+						// "username_template": {
```
Comment (vinay-gopalan) [inline vault/resource_database_secret_backend_connection.go:237]: This can be added back in :)

**Review by vinay-gopalan — APPROVED** [review]
@DrFaust92 sorry for the delay, and thank you so much for submitting this PR! I had just one comment about adding the username template back in since that is an available feature after 1.8, but other than that everything looks and works great! :)

Thanks for contributing!

File: vault/resource_database_secret_backend_connection.go (hunk 1 above)
Comment (DrFaust92) [reply vault/resource_database_secret_backend_connection.go]: done! thanks for the review!

**Pull request merged**